src/main/java/marquez/resources/JobResource.java

```diff
@@ -3,56 +3,112 @@
 import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
 
 import com.codahale.metrics.annotation.Timed;
+import java.util.List;
+import java.util.Optional;
+import javax.validation.Valid;
 import javax.ws.rs.Consumes;
-import javax.ws.rs.DefaultValue;
 import javax.ws.rs.GET;
-import javax.ws.rs.POST;
 import javax.ws.rs.PUT;
 import javax.ws.rs.Path;
 import javax.ws.rs.PathParam;
 import javax.ws.rs.Produces;
-import javax.ws.rs.QueryParam;
+import javax.ws.rs.client.Entity;
 import javax.ws.rs.core.Response;
-import marquez.api.Job;
-import marquez.dao.deprecated.JobDAO;
+import marquez.api.CreateJobRequest;
+import marquez.api.ListJobsResponse;
+import marquez.core.exceptions.ResourceException;
+import marquez.core.exceptions.UnexpectedException;
+import marquez.core.mappers.ApiJobToCoreJobMapper;
+import marquez.core.mappers.CoreJobToApiJobMapper;
+import marquez.core.models.Job;
+import marquez.core.services.JobService;
+import marquez.core.services.NamespaceService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-@Path("/jobs")
+@Path("/api/v1/namespaces/{namespace}/jobs")
 @Produces(APPLICATION_JSON)
 public final class JobResource extends BaseResource {
-  private static final String DEFAULT_LIMIT = "25";
+  private static final Logger LOG = LoggerFactory.getLogger(JobResource.class);
+  private final JobService jobService;
+  private final NamespaceService namespaceService;
 
-  private final JobDAO dao;
+  private ApiJobToCoreJobMapper apiJobToCoreJobMapper = new ApiJobToCoreJobMapper();
+  private CoreJobToApiJobMapper coreJobToApiJobMapper = new CoreJobToApiJobMapper();
 
-  public JobResource(final JobDAO dao) {
-    this.dao = dao;
+  public JobResource(final NamespaceService namespaceService, final JobService jobService) {
+    this.namespaceService = namespaceService;
+    this.jobService = jobService;
   }
 
-  @POST
+  @PUT
+  @Path("/{job}")
   @Consumes(APPLICATION_JSON)
   @Timed
-  public Response create(final Job job) {
-    dao.insert(job);
-    return Response.created(buildURI(Job.class, job.getName())).build();
+  public Response create(
+      @PathParam("namespace") final String namespace,
+      @PathParam("job") final String job,
+      @Valid CreateJobRequest request)
+      throws ResourceException {
+    try {
+      if (!namespaceExists(namespace)) {
+        return Response.status(Response.Status.NOT_FOUND).build();
+      }
+      Job jobToCreate =
```

Comment:
Create jobRequestToCoreModel mapper (for the future). TODO: File an issue.

Reply:
is there an issue for this? can you put the issue # here ?